src/Middleware/UserPageDecorationMiddleware.php

```diff
@@ -52,11 +54,51 @@ public function process(ServerRequestInterface $request, RequestHandlerInterface
                      'test_name' => $this->PageName,
                     );
 
+        $menu    = [];
+        $modules = \Module::getActiveModules($this->DB);
+        foreach ($modules as $module) {
+            if (!$module->hasAccess($user)) {
+                continue;
+            }
+
+            $items = $module->getMenuItems();
+
+            if (!empty($items)) {
+                foreach ($items as $menuitem) {
+                    $menu[$menuitem->getCategory() ?? 'Other'][] = $menuitem;
+                }
+            }
+        }
+
+        // The order of the menu categories is currently unsorted. We sort them
+        // alphabetically by key to impose an order. As a hack to avoid reordering
+        // menu for users of existing LORIS instances, we move the 'Admin' category
+        // to the end even though it's alphabetically first (we do this by removing
+        // the key before sorting and then adding it back afterwards.)
+        //
+        // Coincidentally, this almost-alphabetic sorting results in the exact same
+        // order that LORIS previously had.
+        $adminmenu = $menu['Admin'];
+        unset($menu['Admin']);
+        ksort($menu);
+        $menu['Admin'] = $adminmenu;
+
+        // We unconditionally sort within each section because the within-menu
+        // order has never been stable in LORIS, and sorting adds some consistency.
+        foreach ($menu as $cat => $val) {
```

Comment:
$val = is_null($val) ? array() : $val; 
It will fix the first failure.
PHP warning will be printed out in Travis. 
PHP Warning:  usort() expects parameter 1 to be array, null given in /var/www/Loris/src/Middleware/UserPageDecorationMiddleware.php on line 93

Reply:
thanks, fixed